spec/controllers/reviews_controller_spec.rb

```diff
@@ -254,49 +254,94 @@ def valid_session
+    context 'as an admin' do
+      let(:admin_user) { create(:admin_user) }
+      before(:each) do
+        set_current_user admin_user
       end
 
-      it "assigns the requested review as @review" do
-        review = Review.create! valid_attributes
-        put :update, {:id => review.to_param, :review => valid_attributes}, valid_session
-        assigns(:review).should eq(review)
+      describe "with valid params" do
+        it "updates the requested review" do
+          review = Review.create! valid_attributes
+          # Assuming there are no other reviews in the database, this
+          # specifies that the Review created on the previous line
+          # receives the :update_attributes message with whatever params are
+          # submitted in the request.
+          Review.any_instance.should_receive(:update_attributes).with({'these' => 'params'})
+          put :update, {:id => review.to_param, :review => {'these' => 'params'}}, valid_session
+        end
+
+        it "assigns the requested review as @review" do
+          review = Review.create! valid_attributes
+          put :update, {:id => review.to_param, :review => valid_attributes}, valid_session
+          assigns(:review).should eq(review)
+        end
+
+        it "redirects to the review" do
+          review = Review.create! valid_attributes
+          put :update, {:id => review.to_param, :review => valid_attributes}, valid_session
+          response.should redirect_to(review)
+        end
       end
 
-      it "redirects to the review" do
-        review = Review.create! valid_attributes
-        put :update, {:id => review.to_param, :review => valid_attributes}, valid_session
-        response.should redirect_to(review)
+      describe "with invalid params" do
+        it "assigns the review as @review" do
+          review = Review.create! valid_attributes
+          # Trigger the behavior that occurs when invalid params are submitted
+          Review.any_instance.stub(:save).and_return(false)
+          put :update, {:id => review.to_param, :review => {}}, valid_session
+          assigns(:review).should eq(review)
+        end
+
+        it "re-renders the 'edit' template" do
+          review = Review.create! valid_attributes
+          # Trigger the behavior that occurs when invalid params are submitted
+          Review.any_instance.stub(:save).and_return(false)
+          put :update, {:id => review.to_param, :review => {}}, valid_session
+          response.should render_template("edit")
+        end
       end
     end
 
-    describe "with invalid params" do
-      it "assigns the review as @review" do
-        review = Review.create! valid_attributes
-        # Trigger the behavior that occurs when invalid params are submitted
-        Review.any_instance.stub(:save).and_return(false)
-        put :update, {:id => review.to_param, :review => {}}, valid_session
-        assigns(:review).should eq(review)
+    context 'as a user' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.